### PR TITLE
Support click listeners on link component

### PR DESCRIPTION
**Pull request by bradlc**
👋 This PR adds support for click listeners on the `<Link />` component:

```js
import { Link } from 'inertia-react'

function MyComponent() {
  return (
    <Link href="/" onClick={() => {
      // this will now fire on click!
      window.alert('Hello, world!')
    }}>
      Home
    </Link>
  )
}
```

### CODE REVIEW

File: src/Link.js [hunk 1]
```diff
@@ -7,9 +7,14 @@ export default function InertiaLink({
...
   replace = false,
   children,
+  onClick,
   ...props
 }) {
   const visit = useCallback(event => {
+    if (onClick) {
+      onClick(event)
+    }
+
```
Comment (sebastiandedeyne) [inline src/Link.js]: Could you add an early return here with an `e.defaultPrevented` check? That way you can cancel the navigation if desired.

**Comment by reinink** [comment]
I'm curious, if you want a custom click handler, why not just use a regular anchor link? What's the benefit of still using an Inertia link?

File: src/Link.js (hunk 1 above)
Comment (bradlc) [reply src/Link.js]: This is [already handled by `shouldIntercept`](https://github.com/inertiajs/inertia/blob/0c53e248cbeb669cfba3afc202fea18eed639441/src/should-intercept.js#L4)

**Comment by bradlc** [comment]
> I'm curious, if you want a custom click handler, why not just use a regular anchor link? What's the benefit of still using an Inertia link?

If I use a regular anchor link I don’t get Inertia routing, unless I’m misunderstanding what you mean?

This is for when you want the behaviour of the Inertia link component but need to do something else on click _as well_.

**Comment by sebastiandedeyne** [comment]
Thanks!

**Pull request merged**